src/shared/cloudformation/cloudformation.ts

```diff
@@ -747,3 +760,160 @@ export namespace CloudFormation {
+    const relPath = normalizeSeparator(getWorkspaceRelativePath(path) ?? path)
+    const schemas: { [key: string]: string | string[] | undefined } | undefined = config.get('schemas')
+    const writeTo = type === 'cfn' ? paths.cfnSchema : paths.samSchema
+    const deleteFrom = type === 'sam' ? paths.cfnSchema : paths.samSchema
+    let newWriteArr: string[] = [relPath]
+    let newDeleteArr: string[] = []
+
+    if (schemas) {
+        if (schemas[writeTo]) {
+            newWriteArr = Array.isArray(schemas[writeTo])
+                ? (schemas[writeTo] as string[])
+                : [schemas[writeTo] as string]
+            if (!newWriteArr.includes(relPath)) {
+                newWriteArr.push(relPath)
+            }
+        }
+        if (schemas[deleteFrom]) {
+            const temp = Array.isArray(schemas[deleteFrom])
+                ? (schemas[deleteFrom] as string[])
+                : [schemas[deleteFrom] as string]
+            newDeleteArr = temp.filter(val => val !== relPath)
+        }
+    }
+
+    try {
+        await config.update(
+            'schemas',
+            {
+                ...(schemas ? schemas : {}),
+                [writeTo]: newWriteArr,
+                [deleteFrom]: newDeleteArr,
+            }
+        )
+    } catch (e) {
+        getLogger().error('Could not write YAML schemas to configuration')
+    }
+    
+}
+
+/**
+ * Parses Goformation manifest and generates object containing latest version and schema URLs 
+ * @param manifest Manifest JSON from Github
+ */
+export function getManifestDetails(manifest: string): { samUrl: string; cfnUrl: string; version: string } {
+    const json = JSON.parse(manifest)
+    if (json.tag_name) {
+        return {
+            samUrl: `https://raw.githubusercontent.com/awslabs/goformation/${json.tag_name}/schema/sam.schema.json`,
+            cfnUrl: `https://raw.githubusercontent.com/awslabs/goformation/${json.tag_name}/schema/cloudformation.schema.json`,
+            version: json.tag_name,
+        }
+    } else {
+        throw new Error('Manifest did not include a tag_name')
+    }
+}
+
+/**
+ * Pulls a remote version of file if the local version doesn't match the manifest version (does not check semver increases) or doesn't exist
+ * Pulls local version of file if it does. Uses remote as baskup in case local doesn't exist
+ * @param params.filepath Path to local file
+ * @param params.version Remote version
+ * @param params.url Url to fetch from
+ * @param params.cacheKey Cache key to check version against
+ */
+export async function getRemoteOrCachedFile(params: {
+    filepath: string
+    version: string
+    url: string
+    cacheKey: string
+}): Promise<string | undefined> {
+    const dir = path.parse(params.filepath).dir
+    if (!(await filesystemUtilities.fileExists(dir))) {
+        mkdirSync(dir, { recursive: true })
+    }
+    const cachedVersion = vscode.workspace.getConfiguration(extensionSettingsPrefix).get<string>(params.cacheKey)
```

Comment:
We're super inconsistent. We have `DefaultSettingsConfiguration` (could probably use this one), an abstraction in `workspace` (for spoofing), and a lot of places where we use it directly. We can converge but I'd say this is out of scope for this PR and should be done globally instead.

Reply:
Since there is an inconsistency I'll agree that it is out-of-scope, though moving forward we should push for `DefaultSettingsConfiguration` in my opinion. Centralized control of setting configuration would be extremely powerful when we need to deprecate settings.

Reply:
>when we need to deprecate settings.

This isn't quite the purpose, I think this was essentially another testing shim/type enforcer. Deprecating settings would be done through the `package.json` file, and that's only for `set` calls on newer versions of the extension host (it'll yell at you if an extension doesn't provide the setting in question). You can always read from undocumented settings; a good example of this is the `forceCloud9` setting.

Reply:
I think I misspoke and said deprecate instead migrate but I guess they're pretty similar in their effects.